### PR TITLE
Added gdb tests in CI

### LINKED ISSUE #1189: Run GDB tests for numba-dpex on public GitHub Actions
We want to run [numba_dpex/tests/debugging](https://github.com/IntelPython/numba-dpex/tree/main/numba_dpex/tests/debugging) tests on every Pull Request. That tests being run in [.github/workflows/conda-package.yml#L189-L191](https://github.com/IntelPython/numba-dpex/blob/main/.github/workflows/conda-package.yml#L189-L191). We need to add [Oneapi GDB](https://www.intel.com/content/www/us/en/developer/tools/oneapi/distribution-for-gdb.html) into this environment in order to run those tests.

Implementation proposal:
1. After [setting up conda environment](https://github.com/IntelPython/numba-dpex/blob/main/.github/workflows/conda-package.yml#L122-L123) install Oneapi GDB. Since it is not available as a conda package preferred way is to use [standalone version](https://www.intel.com/content/www/us/en/developer/articles/tool/oneapi-standalone-components.html#distributiongdb). Job should run something like:
```bash
wget -nc -q $ONEAPI_INSTALLER_URL/$ONEAPI_INSTALL_BINARY_NAME 
chmod +x $ONEAPI_INSTALL_BINARY_NAME
./$ONEAPI_INSTALL_BINARY_NAME -a -s --eula accept \
  --action install --components $ONEAPI_COMPONENTS  \
  --install-dir $ONEAPI_INSTALL_DIR --log-dir $ONEAPI_LOG_DIR \
  --download-cache $ONEAPI_CACHE_DIR
``` 

Use `--help` instructions to see available option. Also see the output of the install since you need to add installed binaries to [PATH](https://www.linfo.org/path_env_var.html).

2. As my understanding right some of those tests are failing now. Open issue that describes what tests are failing and why, mute them and add link to the issue in the comment that states why they are muted. We must target those issues in separate PRs.

PS: this is how link to installer should look:
https://registrationcenter-download.intel.com/akdlm/IRC_NAS/fcea1bcd-6a63-4849-b304-906ff71dc2c0/l_dpcpp_dbg_p_2023.2.0.49333.sh

### PR DESCRIPTION
**Pull request by roxx30198**
- [x] Have you provided a meaningful PR description?
- [x] Have you added a test, reproducer or referred to an issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

Added gdb tests for CI and mark the failing tests as xfail or skip.
Closes #1189

### CODE REVIEW

**Comment by ZzEeKkAa** [comment]
You have all gdb tests skipped `============================= 40 skipped in 0.84s ==============================`.

It is likely because oneapi gdb environment variables are not transferred between steps. Try running `setenv` in the same step as running gdb tests

File: setup.cfg [hunk 1]
```diff
@@ -13,9 +13,9 @@ tag_prefix =
 
 [tool:pytest]
 xfail_strict=true
-addopts =
```
Comment (ZzEeKkAa) [inline setup.cfg:16]: Please revert these changes

File: .github/workflows/conda-package.yml [hunk 2]
```diff
@@ -98,6 +98,10 @@ jobs:
       run:
         shell: ${{ matrix.os == 'windows-latest' && 'cmd /C CALL {0}' || 'bash -l {0}' }}
 
+    env:
```
Comment (ZzEeKkAa) [inline .github/workflows/conda-package.yml:101]: Please move it step level.

Comment (roxx30198) [reply .github/workflows/conda-package.yml]: Moved it inside the `run gdb` step.

File: .github/workflows/conda-package.yml [hunk 3]
```diff
@@ -105,6 +105,7 @@ jobs:
         os: [ubuntu-20.04, ubuntu-latest, windows-latest]
         experimental: [false]
         use_mlir: [false]
+        run_gdb: [true, false]
```
Comment (ZzEeKkAa) [inline .github/workflows/conda-package.yml:108]: Please add exclude for windows with run_gdb true.
https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#excluding-matrix-configurations

File: .github/workflows/conda-package.yml [hunk 4]
```diff
@@ -209,6 +222,21 @@ jobs:
             python ${script} || exit 1
           done
 
+      - name: Run gdb tests
+        if: ${{runner.os == 'Linux' && matrix.run_gdb}}
```
Comment (ZzEeKkAa) [inline .github/workflows/conda-package.yml:226]:
```suggestion
        if: ${{matrix.run_gdb}}
```

File: .github/workflows/conda-package.yml [hunk 5]
```diff
@@ -194,6 +206,7 @@ jobs:
         run: python -c "import dpnp, dpctl, numba_dpex; dpctl.lsplatform()"
 
       - name: Run tests
+        if: ${{matrix.run_gdb  == false}}
```
Comment (ZzEeKkAa) [inline .github/workflows/conda-package.yml:209]:
```suggestion
        if: ${{matrix.run_gdb == false}}
```
Please try to be consistent with spaces. Some for other PRs and places

Comment (ZzEeKkAa) [reply .github/workflows/conda-package.yml:209]: Does `if: !${{matrix.run_gdb}}` work?

Comment (roxx30198) [reply .github/workflows/conda-package.yml]: no

Comment (roxx30198) [reply .github/workflows/conda-package.yml]: moved ` !`  inside, not sure that will work

File: .github/workflows/conda-package.yml [hunk 6]
```diff
@@ -181,6 +185,17 @@ jobs:
           $script_path="$env:CONDA_PREFIX\Scripts\set-intel-ocl-icd-registry.ps1"
           &$script_path
 
+      - name: Install pytest-cov
+        run: |
+          mamba install pytest-cov
```
Comment (ZzEeKkAa) [inline .github/workflows/conda-package.yml:190]: Please move it to `Install builded package`

Comment (ZzEeKkAa) [reply .github/workflows/conda-package.yml:190]: You can simply change `pytest` to `pytest-cov`

File: .github/workflows/conda-package.yml [hunk 7]
```diff
@@ -194,6 +209,7 @@ jobs:
         run: python -c "import dpnp, dpctl, numba_dpex; dpctl.lsplatform()"
 
       - name: Run tests
+        if: ${{!matrix.run_gdb}}
```
Comment (ZzEeKkAa) [inline .github/workflows/conda-package.yml:212]: we also don't need to run examples.

**Review by ZzEeKkAa — APPROVED** [review]

**Comment by ZzEeKkAa** [comment]
LGTM! Thank you!
